src/energyplus/Test/ScheduleInterval_GTest.cpp

```diff
@@ -277,6 +277,113 @@ TEST_F(EnergyPlusFixture, ForwardTranslator_ScheduleFixedInterval_Hourly_Shifted
+    EXPECT_EQ("Through: 01/05", objects[0].getString(14, false).get());
+    EXPECT_EQ("For: AllDays", objects[0].getString(15, false).get());
+    EXPECT_EQ("Until: 24:00", objects[0].getString(16, false).get());
+    EXPECT_EQ("9.86222222222222", objects[0].getString(17, false).get());
+    EXPECT_EQ("Through: 01/06", objects[0].getString(18, false).get());
+    EXPECT_EQ("For: AllDays", objects[0].getString(19, false).get());
+    EXPECT_EQ("Until: 24:00", objects[0].getString(20, false).get());
+    EXPECT_EQ("9.80444444444445", objects[0].getString(21, false).get());
+  }
+
+  // timestep 2 days
+  {
+    Model model;
+    Date start_date(MonthOfYear::Jan, 1, 2007);
+    Time timestep_2day(2, 0);
+    Vector daily_wh_inlet_temperatures_c(5);
+    daily_wh_inlet_temperatures_c[0] = 10.0422222222222;
+    daily_wh_inlet_temperatures_c[1] = 9.98111111111111;
+    daily_wh_inlet_temperatures_c[2] = 9.92111111111111;
+    daily_wh_inlet_temperatures_c[3] = 9.86222222222222;
+    daily_wh_inlet_temperatures_c[4] = 9.80444444444445;
+    TimeSeries time_series_tmains(start_date, timestep_2day, daily_wh_inlet_temperatures_c, "C");
+
+    boost::optional<ScheduleInterval> scheduleInterval = ScheduleInterval::fromTimeSeries(time_series_tmains, model);
+    ASSERT_FALSE(scheduleInterval);
+  }
+}
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_ScheduleFixedInterval_Hours) {
+  Model model;
+  Date start_date(MonthOfYear::Jan, 1, 2007);
+  Time timestep_hour(0, 1);
+  Vector hourly_wh_inlet_temperatures_c(5);
+  hourly_wh_inlet_temperatures_c[0] = 10.0422222222222;
+  hourly_wh_inlet_temperatures_c[1] = 9.98111111111111;
+  hourly_wh_inlet_temperatures_c[2] = 9.92111111111111;
+  hourly_wh_inlet_temperatures_c[3] = 9.86222222222222;
+  hourly_wh_inlet_temperatures_c[4] = 9.80444444444445;
+  TimeSeries time_series_tmains(start_date, timestep_hour, hourly_wh_inlet_temperatures_c, "C");
+
+  boost::optional<ScheduleInterval> scheduleInterval = ScheduleInterval::fromTimeSeries(time_series_tmains, model);
+  ASSERT_TRUE(scheduleInterval);
+  EXPECT_TRUE(scheduleInterval->optionalCast<ScheduleFixedInterval>());
+  ScheduleFixedInterval scheduleFixedInterval = scheduleInterval->optionalCast<ScheduleFixedInterval>().get();
+  EXPECT_EQ(60, scheduleFixedInterval.intervalLength());  // one hour in minutes
+  EXPECT_EQ(1, scheduleFixedInterval.startMonth());
+  EXPECT_EQ(1, scheduleFixedInterval.startDay());
+
+  // Forward translate the schedule
+  ForwardTranslator ft;
+  Workspace workspace = ft.translateModel(model);
+
+  std::vector<WorkspaceObject> objects = workspace.getObjectsByType(IddObjectType::Schedule_Compact);
+  ASSERT_EQ(1u, objects.size());
+
+  EXPECT_EQ("Through: 01/01", objects[0].getString(2, false).get());
+  EXPECT_EQ("For: AllDays", objects[0].getString(3, false).get());
+  EXPECT_EQ("Until: 01:00", objects[0].getString(4, false).get());
+  EXPECT_EQ("10.0422222222222", objects[0].getString(5, false).get());
```

Comment:
I think this was already working for hourly timestep, but adding a test just to make sure.